src/modules/gnd_pos_control/GroundRoverPositionControl.hpp

```diff
@@ -0,0 +1,212 @@
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in
+ *    the documentation and/or other materials provided with the
+ *    distribution.
+ * 3. Neither the name PX4 nor the names of its contributors may be
+ *    used to endorse or promote products derived from this software
+ *    without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+
+/**
+ *
+ * This module is a modification of the fixed wing module and it is designed for ground rovers.
+ * It has been developed starting from the fw module, simplified and improved with dedicated items.
+ *
+ * All the ackowledgments and credits for the fw wing app are reported in those files.
+ *
+ * @author Marco Zorzi <mzorzi@student.ethz.ch>
+ */
+
+#include <px4_config.h>
+#include <px4_defines.h>
+#include <px4_posix.h>
+#include <px4_tasks.h>
+
+#include <cfloat>
+
+#include <drivers/drv_hrt.h>
+#include <ecl/l1/ecl_l1_pos_controller.h>
+#include <geo/geo.h>
+#include <mathlib/mathlib.h>
+#include <systemlib/perf_counter.h>
+#include <systemlib/pid/pid.h>
+#include <uORB/topics/control_state.h>
+#include <uORB/topics/fw_pos_ctrl_status.h>
+#include <uORB/topics/manual_control_setpoint.h>
+#include <uORB/topics/parameter_update.h>
+#include <uORB/topics/position_setpoint_triplet.h>
+#include <uORB/topics/vehicle_attitude_setpoint.h>
+#include <uORB/topics/vehicle_control_mode.h>
+#include <uORB/topics/vehicle_global_position.h>
+#include <uORB/uORB.h>
+
+using matrix::Dcmf;
+
+class GroundRoverPositionControl
+{
+public:
+	GroundRoverPositionControl();
+	~GroundRoverPositionControl();
+	GroundRoverPositionControl(const GroundRoverPositionControl &) = delete;
+	GroundRoverPositionControl operator=(const GroundRoverPositionControl &other) = delete;
+
+	/**
+	 * Start the sensors task.
+	 *
+	 * @return	OK on success.
+	 */
+	static int	start();
+
+	/**
+	 * Task status
+	 *
+	 * @return	true if the mainloop is running
+	 */
+	bool		task_running() { return _task_running; }
+
+private:
+	orb_advert_t	_mavlink_log_pub{nullptr};
```

Comment:
@dagar this is unused, can we remove it? I also have a commit ready to fix the coding style, this should fix the checks issues once and for all